src/gui/rss/rsswidget.cpp

```diff
@@ -610,6 +612,49 @@ void RSSWidget::renderArticle(const RSS::Article *article) const
 
         html += u"<pre>" + description + u"</pre>";
     }
+
+    // Supplement relative path to absolute path
+    const QUrl url {article->link()};
+    const QString baseUrl = url.toString(QUrl::RemovePath | QUrl::RemoveQuery);
+    convertRelativeUrlToAbsolute(html, baseUrl);
     html += u"</div>";
     m_ui->textBrowser->setHtml(html);
 }
+
+void convertRelativeUrlToAbsolute(QString &html, const QString &baseUrl)
+{
+    const QRegularExpression rx {uR"(((<a\s+[^>]*?href|<img\s+[^>]*?src)\s*=\s*["'])((https?|ftp):)?(\/\/[^\/]*)?(\/?[^\/"].*?)(["']))"_s
+        , QRegularExpression::CaseInsensitiveOption};
+
+    QString normalizedBaseUrl = baseUrl.endsWith(u'/') ? baseUrl : baseUrl + u'/';
+    QRegularExpressionMatchIterator iter = rx.globalMatch(html);
+
+    while (iter.hasNext())
+    {
+        const QRegularExpressionMatch match = iter.next();
+        const QString scheme = match.captured(4);
+        const QString host = match.captured(5);
+        if (!scheme.isEmpty())
+        {
+            if (host.isEmpty())
+                break; // invalid URL, should never happen
+                
+            // already absolute path
+            continue;
+        }
+
+        QString relativePath = match.captured(6);
+        if (relativePath.startsWith(u'/'))
+            relativePath = relativePath.mid(1);
+
+        if (!host.isEmpty())
+            normalizedBaseUrl = u"http:" + host;
```

Comment:
Please explain this piece of logic.

Reply:
For the Scheme-relative URL: `//developer.mozilla.org/en-US/docs/Learn`, try to complete with default http scheme. I'm considering to retrieve scheme from baseUrl.

Reply:
> For the Scheme-relative URL: `//developer.mozilla.org/en-US/docs/Learn`, try to complete with default http scheme. I'm considering to retrieve scheme from baseUrl.

Ok.
But it is unlikely that this host should be used in the next iterations. `normalizedBaseUrl` should not be modified.

Reply:
Right. Fixed this, and made `normalizedBaseUrl` const again.